rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java

```diff
@@ -1841,6 +1890,20 @@ public Response deleteStudentContactLogEntry(@PathParam("STUDENTID") Long studen
     if (!contactLogEntry.getStudent().getId().equals(contactLogEntry.getStudent().getId())) {
       return Response.status(Status.NOT_FOUND).build();
     }
+    
+    User loggedUser = sessionController.getUser();
+    StaffMember staffMember = userController.findStaffMemberById(loggedUser.getId());
+    Boolean amIGuidanceCounselor = studentController.amIGuidanceCouncelor(studentId, staffMember);
+    
+    if (!loggedUser.getRole().equals(Role.ADMINISTRATOR)) {
+      if (!amIGuidanceCounselor) {
+        if (contactLogEntry.getCreatorId() != null) {
+          if (!contactLogEntry.getCreatorId().equals(sessionController.getUser().getId())) {
+            return Response.status(Status.FORBIDDEN).build();
+          }
+        }
+      }
+    }
 
     if (permanent) {
       studentContactLogEntryController.deleteStudentContactLogEntry(contactLogEntry);
```

Comment:
Check both archive and delete methods in `StudentContactLogEntryController`. Both really should archive/delete all comments related to the entry before archiving/deleting the entry itself.